crates/core/transaction/src/transaction.rs

```diff
@@ -7,14 +7,15 @@ use anyhow::{Context, Error};
 use ark_ff::Zero;
 use decaf377::Fr;
 use decaf377_rdsa::{Binding, Signature, VerificationKey, VerificationKeyBytes};
+use penumbra_asset::Balance;
 use penumbra_community_pool::{CommunityPoolDeposit, CommunityPoolOutput, CommunityPoolSpend};
 use penumbra_dex::{
     lp::action::{PositionClose, PositionOpen},
     swap::Swap,
 };
```

Comment:
https://github.com/penumbra-zone/penumbra/blob/ff86c9a2fd502796156c244c66458fcada046603/crates/core/transaction/src/transaction.rs#L735-L739

does this lead to infinite recursion?

Reply:
Yup. good catch

Reply:
fixed the domain type 

```ts
impl From<&Transaction> for pbt::Transaction {
    fn from(msg: &Transaction) -> Self {
        Transaction {
            transaction_body: msg.transaction_body.clone(),
            anchor: msg.anchor.clone(),
            binding_sig: msg.binding_sig.clone(),
        }
        .into()
    }
}
```